src/XMakeBuildEngine/BackEnd/Node/NodeConfiguration.cs

```diff
@@ -142,7 +142,7 @@ public void Translate(INodePacketTranslator translator)
             translator.Translate(ref _nodeId);
             translator.Translate(ref _buildParameters, BuildParameters.FactoryForDeserialization);
             translator.TranslateArray(ref _forwardingLoggers, LoggerDescription.FactoryForTranslation);
-#if FEATURE_APPDOMAIN
+#if FEATURE_BINARY_SERIALIZATION
```

Comment:
What's the reason for this change?  You shouldn't be serializing AppDomainSetup unless AppDomain's are available.  Combining both also seems reasonable:

    #if FEATURE_APPDOMAIN && FEATURE_BINARY_SERIALIZATION